skops/card/_model_card.py

```diff
@@ -0,0 +1,127 @@
+    -------
+    str:
+        Markdown table of hyperparameters.
+    """
+    hyperparameter_dict = model.get_params(deep=True)
+    table = "| Hyperparameters | Value |\n| :-- | :-- |\n"
+    for hyperparameter, value in hyperparameter_dict.items():
+        table += f"| {hyperparameter} | {value} |\n"
+    return table
+
+
+def create_model_card(
+    model,
+    card_data,
+    **card_kwargs,
+):
+    """Creates a model card for the model and saves it to the target directory.
+
+    Parameters:
+    ----------
+    model: estimator
+        scikit-learn pipeline or model.
+    card_data: CardData
+        CardData object. See the
+        [docs](https://github.com/nateraw/modelcards/blob/7cb1c427a75c0a796631c137c5690ceedab3b2f8/modelcards/card_data.py#L78).
+    card_kwargs:
+        Card kwargs are information you can pass to fill in the sections of the
+        card template, e.g. description of model
+    """
+    model_plot = re.sub(r"\n\s+", "", str(estimator_html_repr(model)))
+    hyperparameter_table = _extract_estimator_config(model)
+    card_data.library_name = "sklearn"
+    template_path = card_kwargs.get("template_path")
+    if template_path is None:
+        template_path = "skops/card/default_template.md"
+    card_kwargs["template_path"] = template_path
+    card = ModelCard.from_template(
+        card_data=card_data,
+        hyperparameter_table=hyperparameter_table,
+        model_plot=model_plot,
+        **card_kwargs,
+    )
+
+    return card
+
+
+def permutation_importances(model, X_test, y_test):
+    importances = permutation_importance(
+        model, X_test, y_test, n_repeats=30, random_state=0
+    )
+    importance = ""
+    for i in importances.importances_mean.argsort()[::-1]:
+        if importances.importances_mean[i] - 2 * importances.importances_std[i] > 0:
+            importance += f"{X_test.columns[i]:<8}\n"
+            importance += f"{importances.importances_mean[i]:.3f}"
+            importance += f" +/- {importances.importances_std[i]:.3f}"
+    if importance != "":
+        importance = "Below are permutation importances:\n\n" + importance
+
+    return importance
+
+
+def evaluate(model, X_test, y_test, metric, dataset_type, dataset_name, task_type):
+    """Evaluates the model and returns the score and the metric.
+    Parameters:
+    ----------
+    model: estimator
+        scikit-learn pipeline or model.
+    X_test: pandas.core.series.Series or numpy.ndarray
+        Split consisting of features for validation.
+    y_test: pandas.core.series.Series or numpy.ndarray
+        Split consisting of targets for validation.
+    metric: str or list
+        sklearn metric key or list of sklearn metric keys. See available list of
+        metrics
+        [here](https://scikit-learn.org/stable/modules/model_evaluation.html).
+    dataset_type: str
+        Type of dataset.
+    dataset_name: str
+        Name of dataset.
+    task_type: str
+        Task type. e.g. tabular-regression
```

Comment:
we should list the task types, but I'm not sure why they are repeated here if they are already on top of the model card.

Reply:
They're needed for EvalResults, see https://github.com/nateraw/modelcards/blob/7cb1c427a75c0a796631c137c5690ceedab3b2f8/modelcards/card_data.py#L9 it's required, I wanted to handle `EvalResults` in here if it's okay.

Reply:
Permalinks and links to code are really bad practice. Code is not written for the end user and we shouldn't expect end users to go read some code on some repo.